src/ir/effects.h

```diff
@@ -110,28 +111,34 @@ struct EffectAnalyzer
     return globalsRead.size() + globalsWritten.size() > 0;
   }
   bool accessesMemory() const { return calls || readsMemory || writesMemory; }
-  bool transfersControlFlow() const { return branches || throws; }
+  // Check whether this may transfer control flow to somewhere outside of this
+  // expression (aside from just flowing out normally). That includes a break
+  // or a throw (if a throw is not inside a catch scope then it cannot transfer
+  // control flow inside the function, which would be interesting info for
+  // some passes, but that requires looking at the outer scope too).
```

Comment:
Not sure if I understand this part of the comment. Did you intend `if a throw is inside a catch scope`? And I'm not sure if 'catch scope' is a correct term, because, a call needs to be within a try body, not a catch body, to be caught.

Reply:
Thanks, I rephrased it now.